src/common/transformations/src/transformations/common_optimizations/pull_through_reduce.cpp

```diff
@@ -0,0 +1,245 @@
+                        ov::SeqGen<int64_t, ov::Direction::FORWARD>(end_pos));
+        return result;
+    }
+    return {};
+}
+
+// Update given reshape_input_shape by inserting "1" dimension on the postion represented by axes_to_insert
+std::shared_ptr<ov::opset9::Constant> update_reshape_target_shape(const ov::Shape& reshape_input_shape,
+                                                                  const std::vector<int64_t>& axes_to_insert) {
+    auto result = std::vector<int64_t>(std::begin(reshape_input_shape), std::end(reshape_input_shape));
+    for (const auto& axis : axes_to_insert) {
+        result.insert(std::next(std::begin(result), axis), 1);
+    }
+    return ov::opset9::Constant::create(ov::element::i64, ov::Shape{result.size()}, result);
+}
+
+// Return true if given inputs have some common elements, otherwise return false.
+bool have_same_axes(const std::vector<int64_t>& unsqueeze_axes, const std::vector<int64_t>& reduce_op_axes) {
+    return std::find_first_of(std::begin(unsqueeze_axes),
+                              std::end(unsqueeze_axes),
+                              std::begin(reduce_op_axes),
+                              std::end(reduce_op_axes)) != std::end(unsqueeze_axes);
+}
+}  // namespace
+
+ov::pass::PullUnsqueezeThroughReduce::PullUnsqueezeThroughReduce() {
+    MATCHER_SCOPE(PullUnsqueezeThroughReduce);
+
+    const auto input = pattern::any_input(pattern::has_static_rank());
+    const auto unsqueeze_axes = pattern::wrap_type<opset9::Constant>();
+    const auto unsqueeze = pattern::wrap_type<opset9::Unsqueeze>({input, unsqueeze_axes}, pattern::has_static_rank());
+    const auto reduce_axes = pattern::wrap_type<opset9::Constant>();
+    const auto reduce = pattern::wrap_type<op::util::ArithmeticReductionKeepDims, op::util::LogicalReductionKeepDims>(
+        {unsqueeze, reduce_axes});
+
+    matcher_pass_callback callback = [=](pattern::Matcher& m) {
+        auto& pattern_map = m.get_pattern_value_map();
+        const auto input_node = pattern_map.at(input);
+        const auto reduce_node = pattern_map.at(reduce).get_node_shared_ptr();
+        const auto unsqueeze_node = pattern_map.at(unsqueeze).get_node_shared_ptr();
+        auto unsqueeze_axes_input =
+            std::dynamic_pointer_cast<opset9::Constant>(pattern_map.at(unsqueeze_axes).get_node_shared_ptr());
+        auto reduce_axes_input =
+            std::dynamic_pointer_cast<opset9::Constant>(pattern_map.at(reduce_axes).get_node_shared_ptr());
+
+        if (!unsqueeze_axes_input || !reduce_axes_input || !reduce_node) {
+            return false;
+        }
+
+        auto unsqueeze_axes_val = unsqueeze_axes_input->cast_vector<int64_t>();
+        normalize_axes(unsqueeze_node.get(),
+                       unsqueeze_node->get_output_partial_shape(0).rank().get_length(),
+                       unsqueeze_axes_val);
+        const auto reduce_axes_val = std::dynamic_pointer_cast<op::util::ArithmeticReductionKeepDims>(reduce_node)
```

Comment:
please add check for nullptr

Reply:
already checked in if above:
`if (!unsqueeze_axes_input || !reduce_axes_input || !reduce_node)`